src/Sarif.Converters/CppCheckError.cs

```diff
@@ -133,7 +123,11 @@ public Result ToSarifIssue()
             }
 
             PhysicalLocation lastLocationConverted;
-            if (this.Locations.Length == 1)
+            if (this.Locations.Length == 0)
+            {
+                return result;
```

Comment:
>result [](start = 23, length = 6)

This "early out" is a little hard to spot. I suggest restructuring to:
```
if (this.Locations.Length > 0)
{
    // Do all the location-related stuff.
}
```

Then for bonus points, because this is a long method, I would extract a helper method
```
AddLocationsToResult(IEnumerable<CppCheckLocation> cppCheckLocations, Result result)
```

Which would contain everything between the braces.